File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int word_number(string s);
void grade(int i);

int main(void)
{
    string s = get_string("Text: ");
    grade(word_number(s));
}

//Counts the number of letters, words and sentences then return the index
int word_number(string s)
{
    int i, n_lettre = 0, n_mot = 1, n_phrase  = 0;
    for (i = 0; i < strlen(s); i++)
    {
        if (isalpha(s[i]) != 0)
        {
            n_lettre++;
        }
        else if (isspace(s[i]) != 0)
        {
            n_mot++;
        }
        if (s[i] == 33 || s[i] == 46 || s[i] == 63)
        {
            n_phrase++;
        }
    }
    float L = ((float) n_lettre / n_mot) * 100;
    float S = ((float) n_phrase / n_mot) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    return index;
}

//Determines the grade from the index
void grade(int i)
{
    if (i < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (i >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", i);
    }
}

File: sentimental-readability/readability.py
def main():
    text = input("Text: ")

    # Initializing
    sentences = 0
    words = 1
    letters = 0

    # Incrementing
    for i in text:
        if i in ["!", ".", "?"]:
            sentences += 1
        if i.isalpha():
            letters += 1
        if i == " ":
            words += 1

    # Average number of letters per 100 words in text
    L = (letters / words) * 100

    # Average number of sentences per 100 words in text
    S = (sentences / words) * 100

    # Coleman-Liau index
    CL = round(0.0588 * L - 0.296 * S - 15.8)

    # Condition on CL
    if CL < 1:
        print(f"Before Grade 1")
    elif CL > 16:
        print(f"Grade 16+")
    else:
        print(f"Grade {CL}")


if __name__ == "__main__":
    main()